src/ASM/ASMs3Dmx.h

```diff
@@ -199,6 +199,14 @@ class ASMs3Dmx : public ASMs3D, private ASMmxBase
   //! \param[out] n3 Number of nodes in third (w) direction
   //! \param[in] basis Which basis to return size parameters for
   virtual bool getSize(int& n1, int& n2, int& n3, int basis = 0) const;
+protected:
+  //! \brief Returns the volume in the parameter space for an element.
+  //! \param[in] iel Element index
+  double getParametricVolume(int iel) const;
+  //! \brief Returns boundary face area in the parameter space for an element.
+  //! \param[in] iel Element index
+  //! \param[in] dir Local face index of the boundary face
+  double getParametricArea(int iel, int dir) const;
```

Comment:
I know, but I prefer to declare virtual in the subclass also, such that it is clear that it is indeed virtual, without looking at the parent class definition.